### PR TITLE
Add 1.0.3 images

### PR DESCRIPTION
**Pull request by natemcmaster**
Marked as WIP until these TODOs are done:

- [x] Verify 1.0.3 optimized package cache
- [x] Update base microsoft/dotnet-nightly => microsoft/dotnet
- [x] Restore from nuget.org when packages have been published

Worth noting

- this moves `microsoft/aspnetcore:lts` => `microsoft/aspnetcore:1.0.3`
- this moves `microsoft/aspnetcore-build:lts` => `microsoft/aspnetcore-build:1.0.3-projectjson`
- there is no `microsoft/aspnetcore-build:1.0.3` image. Until MSBuild tooling RTMs, build image tags fork on the flavor of SDK included, e.g. `1.0.3-projectjson` and `1.0.3-msbuild`.

cc @cesarbs @glennc @MichaelSimons

### CODE REVIEW

**Comment by natemcmaster** [comment]
Removed wip: label. This is updated and ready to merge. @MichaelSimons could you take a took?

File: 1.1.0/jessie/build-projectjson/README.md [hunk 1]
```diff
@@ -11,8 +11,10 @@ This repository contains images that are used to compile/publish ASP.NET Core ap
 
 - [`1.1.0-projectjson`, `latest` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/master/1.1.0/jessie/build-projectjson/Dockerfile)
```
Comment (MichaelSimons) [inline 1.1.0/jessie/build-projectjson/README.md]: I wish every Docker README.md would sort in descending order however this is not the norm.  What is the reason for being different?

File: 1.1.0/jessie/build-projectjson/README.md [hunk 2]
```diff
@@ -11,8 +11,10 @@ This repository contains images that are used to compile/publish ASP.NET Core ap
...
 - [`1.1.0-msbuild` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/master/1.1.0/jessie/build-msbuild/Dockerfile)
+- [`1.0.3-msbuild`, (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/master/1.0.3/jessie/build-msbuild/Dockerfile)
+- [`1.0.3-projectjson`, `lts` (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/master/1.0.3/jessie/build-projectjson/Dockerfile)
 - [`1.0.1-msbuild`, (*Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/master/1.0.1/jessie/build-msbuild/Dockerfile)
```
Comment (MichaelSimons) [inline 1.1.0/jessie/build-projectjson/README.md]: What is the purpose of continuing to document 1.0.1 artifacts and keep the Dockerfile around?  Do you plan to service these old images?

File: 1.1.0/jessie/build-projectjson/README.md (hunk 1 above)
Comment (natemcmaster) [inline 1.1.0/jessie/build-projectjson/README.md]: I'm not sure how deliberate the sort order is. @glennc is the original author and he chose descending. I like it too as it makes the 'latest' tag more prominent.

File: 1.1.0/jessie/build-projectjson/README.md (hunk 2 above)
Comment (natemcmaster) [inline 1.1.0/jessie/build-projectjson/README.md]: No plans to service the image. Will remove from the readme.

File: 1.1.0/jessie/build-projectjson/README.md (hunk 1 above)
Comment (glennc) [inline 1.1.0/jessie/build-projectjson/README.md]: I didn't have a strong reason for it, I just did what seemed to make sense without spending any time looking at others ordering.

**Comment by natemcmaster** [comment]
:up: :date: Squashed and read to go pending approval from @MichaelSimons

**Comment by natemcmaster** [comment]
> the old assets should be removed

We should have more discussion on this. Will start a thread.

**Pull request merged**